modules/table/src/test/java/org/apache/ignite/internal/table/distributed/raft/PartitionCommandListenerTest.java

```diff
@@ -462,6 +464,48 @@ public void testSafeTime() {
         applySafeTimeCommand(SafeTimeSyncCommand.class, testClock.now());
     }
 
+    @Test
+    void testBuildIndexCommand() {
+        UUID indexId = UUID.randomUUID();
+
+        doNothing().when(storageUpdateHandler).buildIndex(eq(indexId), any(List.class), anyBoolean());
+
+        List<UUID> rowUuids0 = List.of(UUID.randomUUID());
+        List<UUID> rowUuids1 = List.of(UUID.randomUUID());
+        List<UUID> rowUuids2 = List.of(UUID.randomUUID());
+
+        InOrder inOrder = inOrder(partitionDataStorage, storageUpdateHandler);
+
+        commandListener.handleBuildIndexCommand(createBuildIndexCommand(indexId, rowUuids0, false), 10, 1);
+
+        inOrder.verify(partitionDataStorage).lastApplied(10, 1);
+        inOrder.verify(storageUpdateHandler).buildIndex(indexId, rowUuids0, false);
+
+        commandListener.handleBuildIndexCommand(createBuildIndexCommand(indexId, rowUuids1, true), 20, 2);
+
+        inOrder.verify(partitionDataStorage).lastApplied(20, 2);
+        inOrder.verify(storageUpdateHandler).buildIndex(indexId, rowUuids1, true);
+
+        commandListener.handleBuildIndexCommand(createBuildIndexCommand(indexId, rowUuids2, false), 5, 1);
```

Comment:
Why to you pass smaller values here? Shouldn't such command be ignored?

Reply:
Add comment.